de-lib/src/main/java/org/iplantc/de/teams/client/presenter/EditTeamPresenterImpl.java

```diff
@@ -0,0 +1,302 @@
+    }
+
+    void addMembersToTeam(Group group, IsHideable hideable) {
+        progressDlg.updateProgress();
+        Subject self = createSelfSubject();
+        List<Privilege> privileges = view.getMemberPrivileges();
+        List<Subject> membersToAdd = getSubjectsFromPrivileges(privileges);
+        membersToAdd.add(self);
+        serviceFacade.addMembersToTeam(group, membersToAdd, new AsyncCallback<List<UpdateMemberResult>>() {
+            @Override
+            public void onFailure(Throwable throwable) {
+                ErrorHandler.post(throwable);
+                mode = EditTeamView.MODE.EDIT;
+                view.unmask();
+                progressDlg.hide();
+            }
+
+            @Override
+            public void onSuccess(List<UpdateMemberResult> updateMemberResults) {
+                hideable.hide();
+                progressDlg.updateProgress();
+                view.unmask();
+                ensureHandlers().fireEvent(new TeamSaved(group));
+            }
+        });
+    }
+
+    void addPrivilegesToTeam(Group group, List<Privilege> privileges, IsHideable hideable) {
+        progressDlg.updateProgress();
+
+        List<UpdatePrivilegeRequest> updateList = convertPrivilegesToUpdateRequest(privileges);
+
+        UpdatePrivilegeRequestList allUpdates = factory.getUpdatePrivilegeRequestList().as();
+        allUpdates.setRequests(updateList);
+
+        serviceFacade.updateTeamPrivileges(group, allUpdates, new AsyncCallback<List<Privilege>>() {
+            @Override
+            public void onFailure(Throwable throwable) {
+                ErrorHandler.post(throwable);
+                mode = EditTeamView.MODE.EDIT;
+                view.unmask();
+                progressDlg.hide();
+            }
+
+            @Override
+            public void onSuccess(List<Privilege> privileges) {
+                addMembersToTeam(group, hideable);
+            }
+        });
+    }
+
+    Subject createSelfSubject() {
+        Subject subject = factory.getSubject().as();
+        subject.setId(userInfo.getUsername());
+        return subject;
+    }
+
+    @Override
+    public void onUserSearchResultSelected(UserSearchResultSelected userSearchResultSelected) {
+        Subject subject = userSearchResultSelected.getSubject();
+        String tag = userSearchResultSelected.getTag();
+        Privilege privilege = factory.getPrivilege().as();
+        privilege.setSubject(subject);
+        privilege.setPrivilegeType(PrivilegeType.read);
+
+        if (SEARCH_MEMBERS_TAG.equals(tag)) {
+            view.addMembers(Lists.newArrayList(privilege));
+        } else {
+            view.addNonMembers(Lists.newArrayList(privilege));
+        }
+
+    }
+
+    @Override
+    public void onRemoveMemberPrivilegeSelected(RemoveMemberPrivilegeSelected event) {
+        Privilege privilege = event.getPrivilege();
+        if (EditTeamView.MODE.CREATE == mode) {
+            view.removeMemberPrivilege(privilege);
```

Comment:
My intention was that there are 2 cases where a user is in this dialog - creating a team for the first time or editing a team.  
If you're creating a team, add a member, and then decide to delete the member, that's purely within the UI.  No members or privileges have actually been added to the backend, so we just remove the line from the liststore.
If you're editing a team, then 2 service calls are made sequentially - first to remove the selected user as a member and then remove their privileges.  But those changes are in the next PR for editing teams 😄   This PR is all just for creating a team, which I know is a bit confusing.